File: app/src/main/java/com/frankly/randomdog/sharedpreferences/SharedPref.java
package com.frankly.randomdog.sharedpreferences;

import android.content.Context;
import android.content.SharedPreferences;

import com.frankly.randomdog.R;
import com.frankly.randomdog.model.DogModel;
import com.frankly.randomdog.randomdogapp.RandomDogApplication;
import com.google.gson.Gson;
import com.google.gson.reflect.TypeToken;

import java.lang.reflect.Type;
import java.util.ArrayList;

import static android.content.Context.MODE_PRIVATE;


/**
 * @author: ArshVardhanAtreya <arshvardhan@yahoo.co.in>
 */


public class SharedPref {

    private Context context;

    public SharedPref(Context context) {
        this.context = context;
    }

    public void saveDogsData(ArrayList<DogModel> list) {
        SharedPreferences sharedPreferences = context.
                getSharedPreferences(RandomDogApplication.getContext().
                        getString(R.string.pref_name), MODE_PRIVATE);
        SharedPreferences.Editor editor = sharedPreferences.edit();
        Gson gson = new Gson();
        String json = null;
        if (list != null && list.size() > 0) {
            json = gson.toJson(list);
        }
        editor.putString(RandomDogApplication.getContext().getString(R.string.dogs_list), json);
        editor.apply();
    }

    public ArrayList<DogModel> loadDogsData() {
        SharedPreferences sharedPreferences = context.
                getSharedPreferences(RandomDogApplication.getContext().
                        getString(R.string.pref_name), MODE_PRIVATE);
        Gson gson = new Gson();
        String json = sharedPreferences.getString(RandomDogApplication.getContext()
                .getString(R.string.dogs_list), null);
        Type type = new TypeToken<ArrayList<DogModel>>() {
        }.getType();
        return gson.fromJson(json, type);
    }

}

File: app/src/main/java/com/randomdog/sharedpreferences/SharedPref.kt
package com.randomdog.sharedpreferences

import android.content.Context
import com.google.gson.Gson
import com.google.gson.reflect.TypeToken
import com.randomdog.R
import com.randomdog.model.DogModel
import com.randomdog.randomdogapp.RandomDogApplication
import java.util.*

/**
 * @author: ArshVardhanAtreya <arshvardhan></arshvardhan>@yahoo.co.in>
 */
class SharedPref(private val context: Context?) {
    fun saveDogsData(list: ArrayList<DogModel?>?) {
        val sharedPreferences = context?.getSharedPreferences(RandomDogApplication.getContext()!!.getString(R.string.pref_name), Context.MODE_PRIVATE)
        val editor = sharedPreferences?.edit()
        val gson = Gson()
        var json: String? = null
        if (list != null && list.size > 0) {
            json = gson.toJson(list)
        }
        editor?.putString(RandomDogApplication.getContext()?.getString(R.string.dogs_list), json)
        editor?.apply()
    }

    fun loadDogsData(): ArrayList<DogModel?>? {
        val sharedPreferences = context?.getSharedPreferences(RandomDogApplication.getContext()?.getString(R.string.pref_name), Context.MODE_PRIVATE)
        val gson = Gson()
        val json = sharedPreferences?.getString(RandomDogApplication.getContext()
                ?.getString(R.string.dogs_list), null)
        val type = object : TypeToken<ArrayList<DogModel?>?>() {}.type
        return gson.fromJson(json, type)
    }

}